### ISSUE TITLE
SceneShaderGLES2: Program linking failed

### ISSUE THREAD

**Issue by madmiraal** [godotengine/godot#50225]
### Godot version

3.4.beta 009aa63a5723eec59256a2bcd8ee2a038d1aa958

### System information

Android emulator Pixel 5.0 1080x1920 420dpi running Android 11.0 x86

### Issue description

Running an Android project generates the following errors:
```
E/godot: ERROR: SceneShaderGLES2: Program linking failed:
    Error: uniform view_index specified with different precision in different shaders.
E/godot:    at: _display_error_with_code (drivers/gles2/shader_gles2.cpp:126) - SceneShaderGLES2: Program linking failed:
    Error: uniform view_index specified with different precision in different shaders.
E/godot: ERROR: Method failed. Returning: nullptr
E/godot:    at: get_current_version (drivers/gles2/shader_gles2.cpp:422) - Method failed. Returning: nullptr
E/godot: ERROR: Condition "!version" is true. Returned: false
E/godot:    at: bind (drivers/gles2/shader_gles2.cpp:87) - Condition "!version" is true. Returned: false
```

### Steps to reproduce

Run an Android project and view the logcat errors.

### Minimal reproduction project

Any project will do, but here is a simple one:
[HelloAndroid.zip](https://github.com/godotengine/godot/files/6707817/HelloAndroid.zip)

**Comment by Calinou** [comment]
Can you reproduce this in 3.3.2?

**Comment by madmiraal** [comment]
> Can you reproduce this in 3.3.2?

No. It's fine in 3.3.2.

**Comment by akien-mga** [comment]
Likely a bug caused by #48011, CC @BastiaanOlij.

**Comment by BastiaanOlij** [comment]
I think the problem is caused by having this in the vertex shader (`drivers/gles2/shaders/scene.glsl`):
```
#ifdef USE_GLES_OVER_GL
#define lowp
#define mediump
#define highp
#else
precision highp float;
precision highp int;
#endif
```

And this in the fragment shader:
```
#ifdef USE_GLES_OVER_GL
#define lowp
#define mediump
#define highp
#else
#if defined(USE_HIGHP_PRECISION)
precision highp float;
precision highp int;
#else
precision mediump float;
precision mediump int;
#endif
#endif
```

This means that the default precision in the vertex shader is `highp` while the default in the fragment shader is either `highp` or `mediump` depending on settings.

As `view_index` is declared as `uniform int view_index;` it will use the default precision.

The easy fix is to change it to `uniform highp int view_index;` but I also feel that the default precision should match between vertex and fragment shader. Not 100% sure about that one though.

**Comment by reduz** [comment]
@BastiaanOlij The ifdef block is intended, highp on shaders on mobile is too much, so unless needed mediump needs to remain default. What most likely happened is that an uniform or a varying did not have the right precision specified on both fragment and vertex.

**Comment by BastiaanOlij** [comment]
> 
> 
> @BastiaanOlij The ifdef block is intended, highp on shaders on mobile is too much, so unless needed mediump needs to remain default. What most likely happened is that an uniform or a varying did not have the right precision specified on both fragment and vertex.

Thanks for clarifying that, I'll add some remarks in there for the next person who scratched their head :)

And indeed, the issue is most likely the missing permission on the `view_index` uniform I introduced so I'll add that in too.

**Comment by akien-mga** [comment]
Fixed by #50237.
